rs-matter/src/secure_channel/busy.rs

```diff
@@ -0,0 +1,73 @@
+/*
+ *
+ *    Copyright (c) 2020-2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+use log::error;
+
+use crate::error::*;
+use crate::respond::ExchangeHandler;
+use crate::transport::exchange::Exchange;
+
+use super::common::{sc_write, OpCode, SCStatusCodes, PROTO_ID_SECURE_CHANNEL};
+
+/// A Secure Channel implementation that is only capable of sending Busy status codes
+///
+/// Use with e.g.
+///
+/// ```ignore
+/// let matter = Matter::new(...);
+///
+/// // ...
+///
+/// let busy_responder = Responder::new("SC Busy Responder", BusySecureChannel::new(), &matter, 200/*ms*/);
+/// busy_responder.run::<10>().await?;
+/// ```
+///
+/// ... to respond with "I'm busy, please try later" status code to all incoming Secure Channel messages, which were
+/// not accepted in time by the actual Secure Channel responder, due to all its handlers being occupied with work.
+pub struct BusySecureChannel(());
+
+impl BusySecureChannel {
+    #[inline(always)]
+    pub const fn new() -> Self {
+        Self(())
+    }
+
+    pub async fn handle(&self, exchange: &mut Exchange<'_>) -> Result<(), Error> {
+        let meta = exchange.recv().await?.meta();
```

Comment:
what is the logic of fetching and buffer processing here? recv() feels like a stateful action of "receive a buffer" however this is also used for checking proto and returning InvalidProto.

On invalid-proto, I expect ComposedExchange to call another handler, however was the exchange recv() already done? The documentation of `recv()` does not seem to document that calling it several times will return the same packet, it only seems to document that "new packet is fetched" and "if packet still pending, error is returned".

Should some documentation be updated for the seemingly common case of "peek" or "fetch_without_consume" for the purposes of reading metadata protocol id?

Reply:
> On invalid-proto, I expect ComposedExchange to call another handler, however was the exchange recv() already done? The documentation of `recv()` does not seem to document that calling it several times will return the same packet, it only seems to document that "new packet is fetched" and "if packet still pending, error is returned".

Hmmm, you might have a point here! That might be a bug. Let me check...

Reply:
> > On invalid-proto, I expect ComposedExchange to call another handler, however was the exchange recv() already done? The documentation of `recv()` does not seem to document that calling it several times will return the same packet, it only seems to document that "new packet is fetched" and "if packet still pending, error is returned".
> 
> Hmmm, you might have a point here! That might be a bug. Let me check...

Fixed now by:
* Adjusting the semantics of `recv` to return the message that might be pending and might be already fetched by `recv_fetch`
* Fixing `ChainedExchangeHandler` to chain the handlers in a way where the proto id to be handled is known apriori